llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -7380,17 +7380,31 @@ bool AArch64DAGToDAGISel::SelectAddrModeIndexedSVE(SDNode *Root, SDValue N,
     return false;
 
   SDValue VScale = N.getOperand(1);
-  if (VScale.getOpcode() != ISD::VSCALE)
+  std::optional<int64_t> MulImm;
+  if (VScale.getOpcode() == ISD::VSCALE) {
+    MulImm = cast<ConstantSDNode>(VScale.getOperand(0))->getSExtValue();
+  } else if (auto C = dyn_cast<ConstantSDNode>(VScale)) {
+    int64_t ByteOffset = C->getSExtValue();
+    constexpr auto SVEBitsPerBlock = AArch64::SVEBitsPerBlock;
+    auto MinVScale = Subtarget->getMinSVEVectorSizeInBits() / SVEBitsPerBlock;
+    auto MaxVScale = Subtarget->getMaxSVEVectorSizeInBits() / SVEBitsPerBlock;
+
+    if (!MaxVScale || MinVScale != MaxVScale || ByteOffset % MaxVScale != 0)
```

Comment:
This might start being a common idiom.  Do you mind adding a helper function to AArch64Subtarget along the lines of `optional<unsigned> getSVEVectorSizeInBits()`?

Reply:
Thanks, that's a good idea. Would you prefer `optional`, or should we follow the logic that `getMinSVEVectorSizeInBits` and `getMaxSVEVectorSizeInBits` already use and return 0 when nothing is known about the limit?

Reply:
I'm happy either way. Feel free to choose the option (pun intended :) ) you prefer.

Reply:
Thanks - I've added this. Please let me know if that's what you had in mind. :)